mmpose/evaluation/metrics/keypoint_partition_metric.py

```diff
@@ -0,0 +1,132 @@
+    @property
+    def dataset_meta(self) -> Optional[dict]:
+        """Optional[dict]: Meta info of the dataset."""
+        return self._dataset_meta
+
+    @dataset_meta.setter
+    def dataset_meta(self, dataset_meta: dict) -> None:
+        """Set the dataset meta info to the metric."""
+        self._dataset_meta = dataset_meta
+        # sigmas required by coco metric have to be split as well
+        for partition_name, keypoint_ids in self.partitions.items():
+            _dataset_meta = deepcopy(dataset_meta)
+            _dataset_meta['num_keypoints'] = len(keypoint_ids)
+            _dataset_meta['sigmas'] = _dataset_meta['sigmas'][keypoint_ids]
+            self.metrics[partition_name].dataset_meta = _dataset_meta
+
+    def process(self, data_batch: Sequence[dict],
+                data_samples: Sequence[dict]) -> None:
+        """Split data samples by partitions, then call metric.process part by
+        part."""
+        parted_data_samples = {
+            partition_name: []
+            for partition_name in self.partitions.keys()
+        }
+        for data_sample in data_samples:
+            for partition_name, keypoint_ids in self.partitions.items():
+                _data_sample = deepcopy(data_sample)
+                _data_sample['pred_instances'][
+                    'keypoint_scores'] = _data_sample['pred_instances'][
+                        'keypoint_scores'][:, keypoint_ids]
+                _data_sample['pred_instances']['keypoints'] = _data_sample[
+                    'pred_instances']['keypoints'][:, keypoint_ids]
+                _data_sample['gt_instances']['keypoints'] = _data_sample[
+                    'gt_instances']['keypoints'][:, keypoint_ids]
+                _data_sample['gt_instances'][
+                    'keypoints_visible'] = _data_sample['gt_instances'][
+                        'keypoints_visible'][:, keypoint_ids]
+
+                # for coco metric
+                if 'raw_ann_info' in _data_sample:
+                    if 'keypoints' in _data_sample['raw_ann_info']:
+                        keypoints = np.array(
+                            _data_sample['raw_ann_info']['keypoints']).reshape(
+                                -1, 3)
+                        keypoints = keypoints[keypoint_ids]
+                        num_keypoints = np.sum(keypoints[:, 2] > 0)
+                        _data_sample['raw_ann_info'][
+                            'keypoints'] = keypoints.flatten().tolist()
+                        _data_sample['raw_ann_info'][
+                            'num_keypoints'] = num_keypoints
+
+                parted_data_samples[partition_name].append(_data_sample)
+
+        for partition_name, metric in self.metrics.items():
+            metric.process(data_batch, parted_data_samples[partition_name])
+
+    def compute_metrics(self, results: list) -> dict:
+        pass
+
+    def evaluate(self, size: int) -> dict:
+        """Overload the evaluate method of the base class.
+
```

Comment:
I think this line of comment can be removed for the sake of simplicity. What do you think?

Reply:
I think the evaluate method was not designed to be overloaded, so I mention it here in case someone don't know. Anyway, of course it can be removed.